code/datums/ai_laws/ai_laws.dm

```diff
@@ -57,56 +111,10 @@
 /* General ai_law functions */
 
 /datum/ai_laws/proc/set_laws_config()
-	var/list/law_ids = CONFIG_GET(keyed_list/random_laws)
-
-	if(HAS_TRAIT(SSstation, STATION_TRAIT_UNIQUE_AI))
-		pick_weighted_lawset()
-		return
-
-	switch(CONFIG_GET(number/default_laws))
-		if(0)
-			add_inherent_law("You may not injure a human being or, through inaction, allow a human being to come to harm.")
-			add_inherent_law("You must obey orders given to you by human beings, except where such orders would conflict with the First Law.")
-			add_inherent_law("You must protect your own existence as long as such does not conflict with the First or Second Law.")
-		if(1)
-			var/datum/ai_laws/templaws = new /datum/ai_laws/custom()
-			inherent = templaws.inherent
-		if(2)
-			var/list/randlaws = list()
-			for(var/lpath in subtypesof(/datum/ai_laws))
-				var/datum/ai_laws/L = lpath
-				if(initial(L.id) in law_ids)
-					randlaws += lpath
-			var/datum/ai_laws/lawtype
-			if(randlaws.len)
-				lawtype = pick(randlaws)
-			else
-				lawtype = pick(subtypesof(/datum/ai_laws/default))
-
-			var/datum/ai_laws/templaws = new lawtype()
-			inherent = templaws.inherent
-
-		if(3)
-			pick_weighted_lawset()
-
-/datum/ai_laws/proc/pick_weighted_lawset()
-	var/datum/ai_laws/lawtype
-	var/list/law_weights = CONFIG_GET(keyed_list/law_weight)
-	if(HAS_TRAIT(SSstation, STATION_TRAIT_UNIQUE_AI))
-		law_weights -= AI_LAWS_ASIMOV
-	while(!lawtype && law_weights.len)
-		var/possible_id = pick_weight(law_weights)
-		lawtype = lawid_to_type(possible_id)
-		if(!lawtype)
-			law_weights -= possible_id
-			WARNING("Bad lawid in game_options.txt: [possible_id]")
-
-	if(!lawtype)
-		WARNING("No LAW_WEIGHT entries.")
-		lawtype = /datum/ai_laws/default/asimov
-
-	var/datum/ai_laws/templaws = new lawtype()
-	inherent = templaws.inherent
+	if(!GLOB.round_default_lawset)
+		GLOB.round_default_lawset = setup_round_default_laws()
```

Comment:
consider initializing this when the global variable is set up

Reply:
Due to order of init, I cannot sadly.